platform/pc/boot/stage1.s

```diff
@@ -75,11 +75,42 @@ dap:
         .lba          dq 1
 
 
+%ifdef DEBUG
+bits 16
+%include "debug.inc"
+bits 32
+%endif
+
 readsectors:
+        mov eax, [dap.sector_count]
+        cmp eax, 0x80
```

Comment:
These still need to be changed from eax to ax, and 0x80 to 0x0080 (see comment below about comparison width).

Reply:
@wjhun upon debugging, we discovered that dap.offset cannot be updated past 128 * 512 bytes (0x10000) due to its width. This seems to imply that the size of the stage2 image cannot exceed 64KB. Would the dap.offset field have to be changed to a long? We tried this, but it triggered a failure in the int 0x13.